### ISSUE TITLE
Flow 2-Job 2: `1b-upsertCases.js` is not updating some fields properly

### ISSUE THREAD

**Issue by jayesse** [OpenFn/unicef-thailand#81]
## Describe the bug
A primero user clicked the `Sync` button in Primero that triggered this Flow 2-1 [job run](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/06361e9d-cb7b-70a2-bb09-28cb6299b8d3) to OpenFn with these case details: 
`"national_id_no": "1209501299178"`
`Primero Case no.:  9ae47f1`

However, when this [Flow 2-2 sync job](https://openfn.org/projects/primero-thailand-io-unicef---staging/jobs/jyzqmq/) is triggered, the following fields do NOT show up in the [Flow 2-2 run log](https://openfn.org/projects/primero-thailand-io-unicef---staging/runs/06361e9e-8831-748d-95f4-2cc588d3cf3a) , and hence NOT updated in Primero.

```
1. spanking_psu_week_14_9e50568 - Missing  from Flow 2-2 run log
i_feel_happy_when_i_play_with__name_of_primary_caregiver_in_plh_programme___8a1d3f8 - Showing 'null' (instead of true)

**Form CQ1-2**
2. i_feel_warm_and_safe_when_i_stay_with__name_of_primary_caregiver_in_plh_programme___30306e2 - 1. I feel warm and safe when I stay with (name of primary caregiver in PLH programme) ---- No value show on Primero but in the JSON file have the value as True = 0 Not true


3. i_feel_happy_when_i_play_with__name_of_primary_caregiver_in_plh_programme___8a1d3f8 - 5. I feel happy when I play with (name of primary caregiver in PLH programme) ---- No value show on Primero but in the JSON file have the value as True = 2 All true
 
4. i_obey__name_of_primary_caregiver_in_plh_programme__and_perform_duties_such_as_going_to_bed_on_time__doing_homework__playing_on_the_phone_at_times__as_he_she_suggest_26850a9 -  7. I obey (name of primary caregiver in PLH programme) and perform duties such as going to bed on time, doing homework, playing on the phone at times. as he/she suggest ------ No value show on Primero but in the JSON file have the value as True = 1 Partially true
this field did not show the value ------ plase_describe_the_reason_from_above_question_d21b8db ----- “Pleases describe the reason from above question”, below the value show on the JSON file

{
                                    	"question": "จากภาพที่เลือกด้านบนนั้น จงบอกว่าเพราะอะไรจึงรู้สึกเช่นนั้น",
                                    	"questionType": NumberInt("1"),
               	                     "answersList": "เสียใจ"

5. how_often_do_you_feel_unhappy__ce672d4 - 3. How often do you feel unhappy? No value on Primero but JSON show True = 3 Very often

6. how_often_do_you_feel_happy__ccb0fd7  -  4. How often do you feel happy? No value on Primero but JSON show True = 2 Often

7.  how_often_do_you_feel_very_happy__3dfaf8a - 5. How often do you feel very happy? System show the value in English but this one on week14 shows correctly in Thai

8.  how_do_you_feel_when_you_go_to_bed_at_night__ecfdd90 - 7. How do you feel when you go to bed at night? Data on both weeks is show the result in English. 

9. How do you feel when you think of (name of primary caregiver in PLH programme)? 0 = True in JSON, but no value shown in Primero
10. How do you feel when you play with(name of primary caregiver in PLH programme)? 2 = True  in JSON, but no value shown in Primero
11. How do you feel when (name of primary caregiver in PLH programme) mentions you? 1 = True  in JSON, but no value shown in Primero
12. How do you feel when (name of primary caregiver in PLH programme) tells you what to do? 1 = True  in JSON, but no value shown in Primero

13.  Data on Week 14 on question no. 6. How do you feel when you are having dinner with your family?  The system did not show value but in JSON file has a value as True = 1

Form CC
14.  The system did not show any value on Primero for both weeks (first week and week14)
15.  Risk Model - ImproveLevel is missing from Primero (Check the Job script).

```

## To Reproduce
1. Run job 1 `/staging/jobs/1-getPatientData.js` with this Message as state.data: https://www.openfn.org/projects/primero-thailand-io-unicef---staging/messages/06361e9d-c2ae-7df2-8a12-a4017c1f9b9a
4. Then run job 2 `staging/jobs/1b-upsertCases.js` with the final state from this Job 1 run. The run will look like this: https://openfn.org/projects/primero-thailand-io-unicef---staging/runs/06361e9e-8831-748d-95f4-2cc588d3cf3a

We want to troubleshoot the missing fields or wrong values loaded to Primero

### expression.js
*Please make changes in the `staging` branch!* ⚠️
Link to the job itself in Github: https://github.com/OpenFn/primero-thailand/blob/staging/jobs/1b-upsertCases.js

#### adaptor
`language-primero@2.9.1`

### state.json
This is step 2 job in a flow. 
1. Run job1 `/staging/jobs/1-getPatientData.js` to generate state for job 2. 
- state.configuration: `MOPH Thailand HIS API - DEMO Alpha (Unicef)`
- state.data: Use this Message: https://openfn.org/projects/primero-thailand-io-unicef---staging/runs/06361e9e-8831-748d-95f4-2cc588d3cf3a
2.  Use the final state of the first job to run the 2nd job
- `state.configuration`: see LP `Primero Thailand Demo API User (Open Function)` (Note... this user is for **basic auth**.)

## Expected behavior
We should see the correct data loaded to Primero. 

## To test/resolve
1. After the desired output is working locally, please push commits to `staging`.
2. I will then re-run [this job run](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/0636249e-ef0a-7377-a3f2-cc3d63d1d697) to re-trigger the flow on platform. 

## Toggl 
UNICEF Thailand

**Comment by mtuchi** [comment]
@jayesse as per our call, I have fixed all the issues marked as checked in this list 👇🏽  on this PR #83, we can sync later for the pending ones

>1. spanking_psu_week_14_9e50568 - Missing  from Flow 2-2 run log 
>- [x] |> Because in the answer { "score": "8+", "answer": "8 ครั้งขึ้นไป", "checked": true  } score is 8+ and not 8 

> i_feel_happy_when_i_play_with__name_of_primary_caregiver_in_plh_programme___8a1d3f8 - Showing 'null' (instead of true) 
>- [ ] |> Question not found in state |> 5. ฉันชอบที่ [[(ระบุชื่อผู้เข้าอบรม PLH):careGiverCalledName]] เล่นกับฉัน

>**Form CQ1-2** 2. i_feel_warm_and_safe_when_i_stay_with__name_of_primary_caregiver_in_plh_programme___30306e2 - 1. I feel warm and safe when I stay with (name of primary caregiver in PLH programme) ---- No value show on Primero but in the JSON file have the value as True = 0 Not true
This shows as null in the API response, as seen in [Line 1326 of the run log](https://openfn.org/projects/primero-thailand-io-unicef---staging/runs/06361e9e-8831-748d-95f4-2cc588d3cf3a). Hence Primero shows the correct value. 
>- [ ] |> Question not found in state |> 3. ฉันรู้สึกอบอุ่น และปลอดภัยเมื่ออยู่กับ [[(ระบุชื่อผู้เข้าอบรม PLH):careGiverCalledName]]

>3. i_feel_happy_when_i_play_with__name_of_primary_caregiver_in_plh_programme___8a1d3f8 - 5. I feel happy when I play with (name of primary caregiver in PLH programme) ---- No value show on Primero but in the JSON file have the value as True = 2 All true 
>- [ ] |> Question not found in state |> 5. ฉันชอบที่ [[(ระบุชื่อผู้เข้าอบรม PLH):careGiverCalledName]] เล่นกับฉัน
 
>4. i_obey__name_of_primary_caregiver_in_plh_programme__and_perform_duties_such_as_going_to_bed_on_time__doing_homework__playing_on_the_phone_at_times__as_he_she_suggest_26850a9 -  7. I obey (name of primary caregiver in PLH programme) and perform duties such as going to bed on time, doing homework, playing on the phone at times. as he/she suggest ------ No value show on Primero but in the JSON file have the value as True = 1 Partially true 
>- [ ] |> Question not found in state |> 7. ฉันเชื่อฟัง [[(ระบุชื่อผู้เข้าอบรม PLH):careGiverCalledName]] และทำหน้าที่ เช่น เข้านอนตามเวลา ทำการบ้าน เล่นโทรศัพท์เป็นเวลา ตามที่ท่านแนะนำ

>this field did not show the value ------ plase_describe_the_reason_from_above_question_d21b8db ----- “Pleases describe the reason from above question”, below the value show on the JSON file
{
    "question": "จากภาพที่เลือกด้านบนนั้น จงบอกว่าเพราะอะไรจึงรู้สึกเช่นนั้น",
    "questionType": NumberInt("1"),
    "answersList": "เสียใจ" 
}
>- [x] |> Fixed will return "เสียใจ"

>5. how_often_do_you_feel_unhappy__ce672d4 - 3. How often do you feel unhappy? No value on Primero but JSON show True = 3 Very often 
>- [x] |> Question not found in state |> 3. บ่อยแค่ไหนที่เธอรู้สึก...ไม่มีความสุขเลย (ตามภาพหมายเลข 1 นี้) |> Correct qn 3. บ่อยแค่ไหนที่เธอรู้สึก...ไม่มีความสุข (ตามภาพหมายเลข 2 นี้)

>6. how_often_do_you_feel_happy__ccb0fd7  -  4. How often do you feel happy? No value on Primero but JSON show True = 2 Often 
>- [x] |> Question not found in state |> 4. บ่อยแค่ไหนที่เธอรู้สึก…มีความสุข (ตามภาพหมaายเลข 3 นี้) |> Correct 4. บ่อยแค่ไหนที่เธอรู้สึก…มีความสุข (ตามภาพหมายเลข 3 นี้)

>7.  how_often_do_you_feel_very_happy__3dfaf8a - 5. How often do you feel very happy? System show the value in English but this one on week14 shows correctly in Thai 
>- [x] |> Looks fine need more info |> "how_often_do_you_feel_very_happy__3dfaf8a": "once_in_a_while_14725b7",

>8.  how_do_you_feel_when_you_go_to_bed_at_night__ecfdd90 - 7. How do you feel when you go to bed at night? Data on both weeks is show the result in English. 
>- [x] |> Looks fine need more info |> "how_do_you_feel_when_you_go_to_bed_at_night__ecfdd90": "unhappy_4d6e375",

>9. How do you feel when you think of (name of primary caregiver in PLH programme)? 0 = True in JSON, but no value shown in Primero 
>- [x] |> Question not found in state |> 8. เธอรู้สึกอย่างไรเวลาที่เธอคิดถึง [[ผู้ดูแลที่เข้าร่วมอบรม:careGiverCalledName]]? |> 8. เธอรู้สึกอย่างไรเวลาที่เธอคิดถึง [[ผู้ดูแลที่เข้าร่วมอบรม:careGiverName]]?

>10. How do you feel when you play with(name of primary caregiver in PLH programme)? 2 = True  in JSON, but no value shown in Primero 
>- [x] |> Question not found in state |> 9. เธอรู้สึกอย่างไรเวลาที่เธอเล่นกับ [[ผู้ดูแลที่เข้าร่วมอบรม:careGiverCalledName]]? |> 9. เธอรู้สึกอย่างไรเวลาที่เธอเล่นกับ [[ผู้ดูแลที่เข้าร่วมอบรม:careGiverName]]?

>11. How do you feel when (name of primary caregiver in PLH programme) mentions you? 1 = True  in JSON, but no value shown in Primero 
>- [x] |> Question not found in state |> 10. เธอรู้สึกอย่างไรเวลาที่ [[ผู้ดูแลที่เข้าร่วมอบรม:careGiverCalledName]] พูดถึงเธอ? |> 10. เธอรู้สึกอย่างไรเวลาที่ [[ผู้ดูแลที่เข้าร่วมอบรม:careGiverName]] พูดถึงเธอ?

>12. How do you feel when (name of primary caregiver in PLH programme) tells you what to do? 1 = True  in JSON, but no value shown in Primero 
>- [x] |> Question not found in state |> 11. เธอรู้สึกอย่างไรเวลาที่ [[ผู้ดูแลที่เข้าร่วมอบรม:careGiverCalledName]] บอกให้เธอทำอะไรต่างๆ? |> 11. เธอรู้สึกอย่างไรเวลาที่ [[ผู้ดูแลที่เข้าร่วมอบรม:careGiverName]] บอกให้เธอทำอะไรต่างๆ?

>13.  Data on Week 14 on question no. 6. How do you feel when you are having dinner with your family?  The system did not show value but in JSON file has a value as True = 1 
>- [ ] |> Question not found |> 6. เธอรู้สึกอย่างไรเวลาที่นั่งกินอาหารเย็นกับครอบครัว?

>Form CC 14.  The system did not show any value on Primero for both weeks (first week and week14)
>- [ ] |> Data available in state for CC |> "questionnaire_code": "CC", "questionnaire_name": "แบบประเมิน  Parent-Child Conflict Tactics Scale (CTSPC-R) (สำหรับเด็กอายุ 4-9 ปี)",

>15.  Risk Model - ImproveLevel is missing from Primero (Check the Job script).
>- [ ] |> The answer in state is "improveLevel": "Unchange" the expected answer is "unchanged"

**Comment by jayesse** [comment]
@mtuchi Thanks for the succinct feedback. As I hand this off to the client, there are two more outstanding issue.

1. The **Caregiver Contact** field is showing the phone number `0452366666`. On [Line 139 of the Flow 2-1 run log](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/06363db2-24df-7b65-a999-b33d02229cc3), the phone number is `0852222222`. Yet somehow, on [Line 1372 of this Flow 2-2 run log](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/06363dcf-a821-74a7-91af-085bf77bed37), it sent `0452366666` to Primero.


2. The **Caregiver Sex** field is not showing in Primero.
See snapshot below:


![image](https://user-images.githubusercontent.com/97878823/199772463-9aa59d0f-9131-4aa6-9514-138543f0464a.png)


In [Line 32 of this Flow 2-1 log](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/06363db2-24df-7b65-a999-b33d02229cc3), we see the field as `"sex": "1",`
In [Line 1374 of this Flow 2-2 log](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/06363dcf-a821-74a7-91af-085bf77bed37), we see the field as `"plh_caregiver_sex_965ffd1": "Male",`

**Comment by jayesse** [comment]
@mtuchi  - One final issue to flag

- [x] `enrolled_in_school_psu_week_14_440dc25` - 1. Is your child currently enrolled in school? In JSON file show True = 1 “No” but on the Primero show Yes.

**Comment by mtuchi** [comment]
@jayesse fixed all these issues 👇🏽, on this PR #88 

> @mtuchi Thanks for the succinct feedback. As I hand this off to the client, there are two more outstanding issue.
> 
> 1. The **Caregiver Contact** field is showing the phone number `0452366666`. On [Line 139 of the Flow 2-1 run log](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/06363db2-24df-7b65-a999-b33d02229cc3), the phone number is `0852222222`. Yet somehow, on [Line 1372 of this Flow 2-2 run log](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/06363dcf-a821-74a7-91af-085bf77bed37), it sent `0452366666` to Primero.
> 2. The **Caregiver Sex** field is not showing in Primero.
>    See snapshot below:
> 
> In [Line 32 of this Flow 2-1 log](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/06363db2-24df-7b65-a999-b33d02229cc3), we see the field as `"sex": "1",` In [Line 1374 of this Flow 2-2 log](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/06363dcf-a821-74a7-91af-085bf77bed37), we see the field as `"plh_caregiver_sex_965ffd1": "Male",`
> - [x] `enrolled_in_school_psu_week_14_440dc25` - 1. Is your child currently enrolled in school? In JSON file show True = 1 “No” but on the Primero show Yes.

**Comment by jayesse** [comment]
@mtuchi Thanks for your feedback. I did some testing and further troubleshooting. Here are my findings.

1. In Primero, I can see that the value for **Caregiver Contact** has been successfully updated.

2. For the **Caregiver sex** field,  `"plh_caregiver_sex_965ffd1": "Male"` was not displayed in Primero. It turns out that in [Lines 1662 and 1663](https://github.com/OpenFn/primero-thailand/blob/staging/jobs/1b-upsertCases.js#L1662-L1663) of the job script, the values `male` and `female` must be in lowercase. (The client's mappings were in uppercase). After I made this change, the value is now displayed correctly in Primero.

3. For the **PSU Week 14  - Enrolled in school** field (`enrolled_in_school_psu_week_14_440dc25`), there was no value showing in Primero. It turns out that in [Lines 2780 and 2783](https://github.com/OpenFn/primero-thailand/blob/staging/jobs/1b-upsertCases.js#L2780-L2783) of the job script, the values in Primero DB are actually `true` and `false`. 

I realised that `Yes` and `No` are only field labels in Primero, and not actual DB values. Hence after I changed the values from Yes/No to true/false in the job, the value is now displayed correctly in Primero.

Thus all three issues are now fully resolved. Thus I will update the client and close this issue.

CC @aleksa-krolls
